controller/node_controller.go

```diff
@@ -179,6 +195,50 @@ func (nc *NodeController) isResponsibleForReplica(obj interface{}) bool {
 	return replica.Spec.NodeID == nc.controllerID
 }
 
+func (nc *NodeController) isResponsibleForSnapshot(obj interface{}) bool {
```

Comment:
NIT: actually this function can directly return `false` if the setting is disabled.